examples/upbound-aws-provider/composite-resources/serverless-examples/object-processor-app/README.md

```diff
@@ -4,12 +4,74 @@ This is a test Go application for serverless examples.
 ## Pre-requisites:
 - Go is installed. Find the installation instructions [here](https://go.dev/doc/install).
 - AWS CLI is installed and configured. Find the installation instructions [here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
+- Docker is installed. Find the installation instructions [here](https://docs.docker.com/engine/install/).
 
-## Build, zip, and upload golang function code to s3
 Navigate to the function code folder
+```shell
+cd ../object-processor-app/
 ```
-cd composite-resources/serverless/app/
+## Option 1: Container
+This option contains commands to build the Go binary, build a Docker image, and create ECR repo and upload the image to it.
+First, set the region, ECR_URL, and IMAGE_NAME as environment variables.
+```shell
+export AWS_REGION=<replace-me-with-aws-region> # this should make aws cli pointing to the region without explicitly passing --region 
+export ECR_URL=$(aws sts get-caller-identity|jq -r ".Account" | tr -d '[:space:]').dkr.ecr.$AWS_REGION.amazonaws.com
+export IMAGE_NAME=<replace-me-with-image-name>
 ```
+Log in to ECR
+```shell
+aws ecr get-login-password | docker login --username AWS --password-stdin $ECR_URL
+```
+Create ECR repository
+```shell
+aws ecr create-repository --repository-name $IMAGE_NAME
+```
+Build the Go binary
+```shell
+GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o main main.go
+```
+Set the `DOCKER_IMAGE` env var, this step is nessasary to properly escape the ':' for the subsequent docker commands
+```shell
+export DOCKER_IMAGE=$ECR_URL/$IMAGE_NAME\:latest
+```
+Build the docker image
+```shell
+docker build -t $DOCKER_IMAGE . 
```

Comment:
We should provide `finch` command in addition of `docker`, I don't have "docker" installed on my Mac (I'm not allowed)
Re finch: https://github.com/runfinch/finch

Reply:
Docker is the de-facto standard, I prefer to keep it to just docker. If you are using flinch or podman, you can extrapolate the relative command. We cannot/should not include every possible option.